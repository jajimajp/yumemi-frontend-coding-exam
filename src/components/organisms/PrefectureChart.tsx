import { CartesianGrid, Line, LineChart, XAxis, YAxis } from 'recharts'
import { BundledComposition, CompositionPerYearLabel } from '@/types/compositionPerYear'
import { Prefecture } from '@/types/prefecture'

type Prop = {
  label: CompositionPerYearLabel
  prefCodesToShow: Prefecture['prefCode'][]
  data: BundledComposition
}

export default function PrefectureChart({ label, prefCodesToShow, data }: Prop) {
  return (
    <LineChart width={800} height={400} data={data} margin={{ left: 80, right: 5, bottom: 50 }}>
      {prefCodesToShow.map((prefCode) => (
        <Line key={prefCode} type="monotone" dataKey={prefCode} />
      ))}
      <CartesianGrid stroke="#ccc" />
      <XAxis dataKey="year" label={{ value: 'å¹´', position: 'bottom' }} />
      <YAxis label={{ value: label, angle: -90, position: 'left', offset: 50 }} />
    </LineChart>
  )
}
