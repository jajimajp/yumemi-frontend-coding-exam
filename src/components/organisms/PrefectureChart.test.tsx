import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import PrefectureChart from './PrefectureChart'

describe('PrefectureChart.tsx', () => {
  test('スナップショットが作成できる', async () => {
    const result = render(<PrefectureChart label="総人口" prefCodesToShow={[]} data={[]} />)
    expect(result).toMatchSnapshot()
  })
})
