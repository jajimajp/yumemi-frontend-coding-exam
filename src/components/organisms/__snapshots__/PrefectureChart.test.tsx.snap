// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrefectureChart.tsx スナップショットが作成できる 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="recharts-wrapper"
        role="region"
        style="position: relative; cursor: default; width: 800px; height: 400px;"
      >
        <svg
          class="recharts-surface"
          height="400"
          style="width: 100%; height: 100%;"
          viewBox="0 0 800 400"
          width="800"
        >
          <title />
          <desc />
          <defs>
            <clippath
              id="recharts1-clip"
            >
              <rect
                height="320"
                width="655"
                x="140"
                y="0"
              />
            </clippath>
          </defs>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="320"
                stroke="#ccc"
                width="655"
                x="140"
                x1="140"
                x2="795"
                y="0"
                y1="0"
                y2="0"
              />
              <line
                fill="none"
                height="320"
                stroke="#ccc"
                width="655"
                x="140"
                x1="140"
                x2="795"
                y="0"
                y1="320"
                y2="320"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="320"
                stroke="#ccc"
                width="655"
                x="140"
                x1="140"
                x2="140"
                y="0"
                y1="0"
                y2="320"
              />
              <line
                fill="none"
                height="320"
                stroke="#ccc"
                width="655"
                x="140"
                x1="795"
                x2="795"
                y="0"
                y1="0"
                y2="320"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="recharts-wrapper"
      role="region"
      style="position: relative; cursor: default; width: 800px; height: 400px;"
    >
      <svg
        class="recharts-surface"
        height="400"
        style="width: 100%; height: 100%;"
        viewBox="0 0 800 400"
        width="800"
      >
        <title />
        <desc />
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="320"
              width="655"
              x="140"
              y="0"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="320"
              stroke="#ccc"
              width="655"
              x="140"
              x1="140"
              x2="795"
              y="0"
              y1="0"
              y2="0"
            />
            <line
              fill="none"
              height="320"
              stroke="#ccc"
              width="655"
              x="140"
              x1="140"
              x2="795"
              y="0"
              y1="320"
              y2="320"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="320"
              stroke="#ccc"
              width="655"
              x="140"
              x1="140"
              x2="140"
              y="0"
              y1="0"
              y2="320"
            />
            <line
              fill="none"
              height="320"
              stroke="#ccc"
              width="655"
              x="140"
              x1="795"
              x2="795"
              y="0"
              y1="0"
              y2="320"
            />
          </g>
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
